src/components/theme/LtiApplyTheme.js

```diff
@@ -28,34 +17,63 @@ export class LtiApplyTheme extends React.Component {
     this.loadTheme()
   }
 
+  componentDidUpdate(prevProps, prevState, snapshot) {
+    if (this.props.url !== prevProps.url) {
+      this.loadTheme()
+    }
+  }
+
   loadTheme = () => {
+    const {version, url} = this.props;
     if (!this.loading && !this.state.theme) {
-      if (this.props.url) {
+      if (url) {
         this.loading = true
-        fetch(this.props.url).then(response => response.json())
+        fetch(url).then(response => response.json())
           .then((response) => {
             // Apply the loaded theme.
+            let newTheme
+            if(version === 'new') {
+              newTheme = this.props.highContrast ? canvasHighContrast : { ...canvas, ...json }
+            } else {
+              newTheme = ApplyTheme.generateTheme('canvas', response)
+            }
             this.setState({
-              theme: ApplyTheme.generateTheme('canvas', response)
+              theme: newTheme
             })
           }).finally(() => this.loading = false)
       }
     }
   }
 
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    if (this.props.url !== prevProps.url) {
-      this.loadTheme()
+  render() {
+
+    const {version, children} = this.props;
+    const {theme} = this.state;
+
+    if (version === 'New') {
+      return  <EmotionThemeProvider theme={theme}>
+          {children}
+      </EmotionThemeProvider>
```

Comment:
Do you know if it's possible to detect the version available in the project using this library and by default "do the right thing", then only fall back to properties to override it?

Reply:
hmmn, I am not sure that is possible, but I might be wrong

Reply:
Ok, don't worry then.